client/src/App.js

```diff
@@ -14,17 +13,40 @@ export default class App extends Component {
   }
 
   addToSavedList = movie => {
+    console.log("savedList:", this.state.savedList);
+    console.log("movie:", movie);
+    console.log("state check", this.state.savedList.includes(movie));
     const savedList = this.state.savedList;
-    savedList.push(movie);
-    this.setState({ savedList });
+    console.log("var check", savedList.includes(movie));
+    // Added logic so the user cannot save a movie twice:
+    if (savedList.includes(movie)) {
```

Comment:
I see what you mean by your commit message about the list allowing a dupe save after leaving and coming back -- so when you check objects for equality/inequality, or when you check to see if an array includes and object, you're testing for equality to the object `reference` and not the object itself. When staying on the same view and trying to save a movie twice, it still has the same reference for what that object is, but when leave and come back React treats it as a different object because it doesn't actually check the inner contents. A possible solution here would be to make an array of all movie titles from the objects already stored in savedList, then check to see if that array includes the new movie.title for your conditional instead. Tricky stuff.